pkg/security/probe/dentry_resolver.go

```diff
@@ -315,7 +332,7 @@ func (dr *DentryResolver) GetNameFromMap(mountID uint32, inode uint64, pathID ui
 	atomic.AddInt64(dr.hitsCounters[metrics.SegmentResolutionTag][metrics.KernelMapsTag], 1)
 
 	cacheKey := PathKey{MountID: mountID, Inode: inode}
-	cacheEntry := PathEntry{Parent: pathLeaf.Parent, Name: pathLeaf.GetName()}
+	cacheEntry := dr.getPathEntryFromPool(pathLeaf.Parent, pathLeaf.GetName())
 	_ = dr.cacheInode(cacheKey, cacheEntry)
```

Comment:
I think we should check the error from the `cacheInode` method, and call `Put()` if the entry was not inserted in the cache.